src/routers/peers.py

```
@@ -331,10 +333,23 @@
+    search: schemas.MessageSearchOptions = Body(
+        ..., description="Message search parameters "
+    ),
     db: AsyncSession = db,
 ):
     """Search a Peer"""
-    stmt = await crud.search(query, workspace_name=workspace_id, peer_name=peer_id)
+    embed_messages_enabled = settings.LLM.EMBED_MESSAGES
+    if search.use_semantic_search and not embed_messages_enabled:
+        raise DisabledException(
+            "Semantic search requires EMBED_MESSAGES flag to be enabled"
+        )
```

Ruff (0.11.9)

Avoid specifying long messages outside the exception class

(TRY003)